docs/tutorials/data_provider.md

```diff
@@ -133,11 +133,13 @@ where
 }
 
 // Usage example:
-let provider = icu_testdata::buffer();
+// While HelloWorldProvider does not need to be cached, more expensive providers, like
+// deserializing BufferProviders, or providers doing I/O, should be.
```

Comment:
```suggestion
// While HelloWorldProvider does not need to be cached, it may be useful to cache results from
// more expensive providers, like deserializing BufferProviders or providers doing I/O.
```